crates/control/src/active_vision.rs

```diff
@@ -146,3 +157,65 @@ fn generate_field_mark_positions(field_dimensions: &FieldDimensions) -> Vec<Poin
         right_own_penalty_box_corner,
     ]
 }
+
+fn next_point_of_interest(
+    current_point_of_interest: PointOfInterest,
+    field_mark_positions: &[Point2<f32>],
+    obstacles: &[Obstacle],
+    parameters: &LookActionConfiguration,
+    robot_to_field: &Isometry2<f32>,
+    ball: Option<&BallState>,
+) -> PointOfInterest {
+    match current_point_of_interest {
+        PointOfInterest::Forward => {
+            let field_mark_of_interest =
+                closest_field_mark_visible(field_mark_positions, parameters, robot_to_field);
+
+            if let Some(field_mark_position) = field_mark_of_interest {
+                PointOfInterest::FieldMark {
+                    absolute_position: robot_to_field * field_mark_position,
+                }
+            } else if let Some(..) = ball {
+                PointOfInterest::Ball
+            } else {
```

Comment:
What do you think about the following to replace the `if else if else` pattern?

```suggestion
            match (field_mark_of_interest, ball) {
                (Some(field_mark_position), _) => PointOfInterest::FieldMark {
                    absolute_position: robot_to_field * field_mark_position,
                },
                (None, Some(_)) => PointOfInterest::Ball,
                (_, _) => {
```

Reply:
Done